src/renderer/base/renderer.cpp

```diff
@@ -1125,37 +1118,23 @@ void Renderer::_PaintOverlay(IRenderEngine& engine,
 {
     try
     {
-        // First get the screen buffer's viewport.
-        auto view = _pData->GetViewport();
-
         // Now get the overlay's viewport and adjust it to where it is supposed to be relative to the window.
-
-        auto srCaView = overlay.region.ToInclusive();
+        auto srCaView = overlay.region.ToExclusive();
         srCaView.Top += overlay.origin.Y;
         srCaView.Bottom += overlay.origin.Y;
         srCaView.Left += overlay.origin.X;
         srCaView.Right += overlay.origin.X;
 
-        // Set it up in a Viewport helper structure and trim it the IME viewport to be within the full console viewport.
-        auto viewConv = Viewport::FromInclusive(srCaView);
-
         gsl::span<const til::rect> dirtyAreas;
         LOG_IF_FAILED(engine.GetDirtyArea(dirtyAreas));
 
         for (const auto& rect : dirtyAreas)
         {
-            // Dirty is an inclusive rectangle, but oddly enough the IME was an exclusive one, so correct it.
-            auto srDirty = rect.to_small_rect();
-            srDirty.Bottom++;
-            srDirty.Right++;
-
-            if (viewConv.TrimToViewport(&srDirty))
+            if (const auto viewDirty = rect & srCaView)
             {
-                auto viewDirty = Viewport::FromInclusive(srDirty);
-
-                for (auto iRow = viewDirty.Top(); iRow < viewDirty.BottomInclusive(); iRow++)
+                for (auto iRow = viewDirty.Top; iRow < viewDirty.Bottom; iRow++)

```

Comment:
This iterates one more time than the previous loop, doesn't it? Because we're using `Bottom` instead of `BottomInclusive()`?